### PR TITLE
libopae: fix metric api invalid input index and name

### PR DESCRIPTION
**Pull request by anandaravuri**
changes:
	-returns error not found if user passes all invlaid index
	 to getmetricbyindex()

       -returns error not found if user passes all invlaid serach
	 strings to getmetricbyname()

### CODE REVIEW

File: libopae/plugins/xfpga/metrics/metrics.c [hunk 1]
```diff
@@ -310,23 +344,33 @@ fpga_result __FPGA_API__ xfpga_fpgaGetMetricsByName(fpga_handle handle,
...
 				continue;
 			}
 
 			result = get_afu_metric_value(handle, &(_handle->fpga_enum_metric_vector),
 				metric_num,
 				&metrics[i]);
 			if (result != FPGA_OK) {
-				FPGA_ERR("Failed to get metric value");
+				FPGA_ERR("Failed to get metric value  for metric = %s", metrics_names[i]);
+				metrics[i].metric_num = 0xFFFF;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics.c]: Will this result in an array index out of bounds somewhere later?

File: libopae/plugins/xfpga/metrics/metrics.c [hunk 2]
```diff
@@ -336,7 +380,8 @@ fpga_result __FPGA_API__ xfpga_fpgaGetMetricsByName(fpga_handle handle,
 							&(_handle->fpga_enum_metric_vector),
 							&metric_num);
 			if (result != FPGA_OK) {
-				FPGA_ERR("Invalid Input Metrics string");
+				FPGA_ERR("Invalid input metrics string= %s", metrics_names[i]);
+				metrics[i].metric_num = 0xFFFF;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics.c]: Will this result in an array index out of bounds somewhere later?

File: libopae/plugins/xfpga/metrics/metrics.c [hunk 3]
```diff
@@ -345,10 +390,20 @@ fpga_result __FPGA_API__ xfpga_fpgaGetMetricsByName(fpga_handle handle,
 							metric_num,
 							&metrics[i]);
 			if (result != FPGA_OK) {
-				FPGA_ERR("Failed to get metric value");
-				goto out_unlock;
+				FPGA_ERR("Failed to get metric value  for metric = %s \n", metrics_names[i]);
+				metrics[i].metric_num = 0xFFFF;
```
Comment (tswhison) [inline libopae/plugins/xfpga/metrics/metrics.c]: Will this result in an array index out of bounds somewhere later?

File: testing/xfpga/test_metrics_c.cpp [hunk 4]
```diff
@@ -257,4 +257,308 @@ TEST_P(metics_c_p, test_metric_04) {
...
+	dfh.next_header_offset = 0x100;
+	dfh.eol = 1;
+	dfh.reserved = 0;
+	dfh.type = 0x1;
+
+
+	printf("------dfh.csr = %lx \n", dfh.csr);
+	EXPECT_EQ(FPGA_OK, xfpga_fpgaWriteMMIO64(handle_, 0, 0x0, dfh.csr));
+
+	EXPECT_EQ(FPGA_OK, xfpga_fpgaWriteMMIO64(handle_, 0, 0x8, 0xf89e433683f9040b));
```
Comment (tswhison) [inline testing/xfpga/test_metrics_c.cpp]: Can this be made agnostic to the type of AFU somehow using the test framework?

File: testing/xfpga/test_metrics_utils_c.cpp [hunk 5]
```diff
@@ -676,6 +676,6 @@ TEST_P(metics_uitls_dcp_c_p, test_metric_utils_14) {
 
 }
 
-INSTANTIATE_TEST_CASE_P(metics_uitls_c, metics_uitls_dcp_c_p, 
+INSTANTIATE_TEST_CASE_P(metrics_uitls_c, metrics_uitls_dcp_c_p,
```
Comment (jelonanX) [inline testing/xfpga/test_metrics_utils_c.cpp]: s/uitls/utils

Comment (anandaravuri) [reply testing/xfpga/test_metrics_utils_c.cpp]: fixed

File: libopae/plugins/xfpga/metrics/afu_metrics.c [hunk 6]
```diff
@@ -198,7 +199,7 @@ fpga_result add_afu_metrics_vector(fpga_metric_vector *vector,
 	snprintf_s_i(group_name, sizeof(group_name), "%x", group_csr.group_id);
 	snprintf_s_i(metric_name, sizeof(metric_name), "%x", metric_csr.counter_id);
 
-	sprintf(qualifier_name, "%s:%x:%x", "AFU", group_csr.group_id, metric_csr.counter_id);
+	sprintf(qualifier_name, "%s:%x", "AFU", group_csr.group_id);
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/afu_metrics.c]: fixed

File: libopae/plugins/xfpga/metrics/metrics.c (hunk 1 above)
Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics.c]: user sends metrics search string as input , if Metric API couldn't found any metric with name.
struct fpga_metric {
	uint64_t metric_num;    // Metric index num
	metric_value value;     // Metric value
};

OPAE API fills struct fpga_metric with invalid index 0xFFF for not found metric by name.

File: testing/xfpga/test_metrics_c.cpp (hunk 4 above)
Comment (anandaravuri) [reply testing/xfpga/test_metrics_c.cpp]: this codes adds AFU  Header & Metric BBB in MMIO region.

File: libopae/plugins/xfpga/metrics/metrics.c (hunk 1 above)
Comment (tswhison) [reply libopae/plugins/xfpga/metrics/metrics.c]: Can we use a #define name instead?

Comment (anandaravuri) [reply libopae/plugins/xfpga/metrics/metrics.c]: fixed

**Pull request merged**